File: src/main/java/com/maxciv/infer/plugin/config/InferPluginSettings.java
package com.maxciv.infer.plugin.config;

import com.maxciv.infer.plugin.process.BuildTools;

/**
 * @author maxim.oleynik
 * @since 27.11.2018
 */
public class InferPluginSettings {

    private String inferPath = "infer";
    private BuildTools buildTool = BuildTools.DEFAULT;

    InferPluginSettings() {
    }

    public String getInferPath() {
        return inferPath;
    }

    public void setInferPath(String inferPath) {
        this.inferPath = inferPath;
    }

    public BuildTools getBuildTool() {
        return buildTool;
    }

    public void setBuildTool(BuildTools buildTool) {
        this.buildTool = buildTool;
    }
}

File: src/main/kotlin/com/maxciv/infer/plugin/config/InferPluginSettings.kt
package com.maxciv.infer.plugin.config

import com.maxciv.infer.plugin.data.ProjectModule
import com.maxciv.infer.plugin.data.report.InferReport
import com.maxciv.infer.plugin.process.BuildTools
import com.maxciv.infer.plugin.process.OperationSystems
import java.util.concurrent.atomic.AtomicInteger

/**
 * @author maxim.oleynik
 * @since 27.11.2018
 */
data class InferPluginSettings(
    var inferPath: String = "infer",
    var inferWorkingDir: String = "./.idea/infer-out",
    var buildTool: BuildTools = BuildTools.DEFAULT,
    var os: OperationSystems = OperationSystems.defineOs(),
    var projectModules: MutableList<ProjectModule> = mutableListOf(),
    var isAutoscrollToSourceEnabled: Boolean = true,
    var isOnSaveAnalyzeEnabled: Boolean = true,
    var isCompileOnModuleAnalysisEnabled: Boolean = true,
    var isCompileOnlyOneModuleOnModuleAnalysisEnabled: Boolean = false,
    var isShortClassNamesEnabled: Boolean = true,
    var aggregatedInferReport: InferReport = InferReport(),

    @Transient val analysisCounter: AtomicInteger = AtomicInteger(0),
    @Transient var lastAnalysisTime: Long = 0,
    @Transient val filesToAnalyse: MutableList<String> = mutableListOf()
)
